train-controller/src/main/java/hu/bme/mit/train/controller/TrainControllerImpl.java

```diff
@@ -2,7 +2,7 @@
 
 import hu.bme.mit.train.interfaces.TrainController;
 
-public class TrainControllerImpl implements TrainController {
+public class TrainControllerImpl extends Thread implements TrainController {
```

Comment:
rather than the whole class being a thread, create an inner class and only make that a thread